src/rgw/rgw_gc.cc

```diff
@@ -364,16 +457,37 @@ int RGWGC::process(int index, int max_secs, bool expired_only,
     int max = 100;
     std::list<cls_rgw_gc_obj_info> entries;
 
-    ret = cls_rgw_gc_queue_list_entries(store->gc_pool_ctx, obj_names[index], marker, max, expired_only, entries, &truncated, next_marker);
-    ldpp_dout(this, 20) <<
+    int ret = 0;
+    if (! transitioned_objects_cache[index]) {
+      ret = cls_rgw_gc_list(store->gc_pool_ctx, obj_names[index], marker, max, expired_only, entries, &truncated, next_marker);
+      ldpp_dout(this, 20) <<
+      "RGWGC::process cls_rgw_gc_list returned with returned:" << ret <<
+      ", entries.size=" << entries.size() << ", truncated=" << truncated <<
+      ", next_marker='" << next_marker << "'" << dendl;
+      if (entries.size() == 0) {
+        std::list<cls_rgw_gc_obj_info> non_expired_entries;
```

Comment:
@cbodley @mattbenjamin :This extra cls_rgw_gc_list is to check for non-expired entries to make sure there are no entries in omap and we can successfully set the 'transitioned state' to true.